test/CMakeLists.txt

```diff
@@ -8,6 +8,11 @@ project(
   LANGUAGES CXX
 )
 
+# build all dynamic libraries and executables together at bin directory
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+
```

Comment:
Imo these variables should be set externally and are not part of the main configuration. E.g. to test installation locally, you could use something like

```bash
cmake -S. -Bbuild -D CMAKE_INSTALL_PREFIX="$(pwd)/root"
```

```suggestion
```

Reply:
That is the crux with windows DLL: they are not found if not installed in path or same directory like the exe.

Reply:
[MSDN reference](https://docs.microsoft.com/en-us/windows/win32/dlls/dynamic-link-library-search-order#search-order-for-desktop-applications).

You can either do this, so your tests work, or explicitly [set the `PATH` env var](https://github.com/friendlyanon/cxx-static-shared-example/blob/master/test/CMakeLists.txt#L15-L27) on the tests' executable targets.

Reply:
Under `Windoof` the path must be extended to every install package. IMHO this make no sense!

With my concept, you my find every program build at `${CMAKE_BINARY_DIR}/bin` and you may run all tests without any environment settings.

Reply:
> Imo these variables should be set externally and are not part of the main configuration. E.g. to test installation locally, you could use something like
> 
> ```shell
> cmake -S. -Bbuild -D CMAKE_INSTALL_PREFIX="$(pwd)/root"
> ```
the `CMAKE_RUNTIME_OUTPUT_DIRECTORY` is independent from `CMAKE_INSTALL_PREFIX`

Reply:
OK, As long the CI build is not run on Windows too, it doesn't matter.